controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -369,3 +377,70 @@ func (p *OpenAPIProductReconciler) desiredPrivateAPISecurity() *capabilitiesv1be
 
 	return privateAPISec
 }
+
+func (p *OpenAPIProductReconciler) desiredOIDCSecurity() *capabilitiesv1beta1.SecuritySpec {
+	if p.openapiCR.Spec.SecurityScheme == nil && p.openapiCR.Spec.SecurityScheme.Type == nil {
```

Comment:
When `p.openapiCR.Spec.SecurityScheme` is nil, so it is `p.openapiCR.Spec.SecurityScheme.Type`. 

This 
```go
if p.openapiCR.Spec.SecurityScheme == nil && p.openapiCR.Spec.SecurityScheme.Type == nil {
```

is equivalent to 

```go
if p.openapiCR.Spec.SecurityScheme == nil  {
```

Reply:
sure,  Thank you. fixed. Working on desiredOIDCSecurity (TODO)